website/docs/ref/enums.doc.js

```diff
@@ -0,0 +1,78 @@
+/* @flow */
+/*
+---
+id: enums
+title: Enums
+permalink: /docs/enums.html
+prev: react.html
```

Comment:
It might be better to make this a single page for all the special types we want to document: `$PropertyType`, `$NonMaybeType`,  and `$Keys`.

Also, we should probably not call these enums because those could be added to JavaScript separately and they'd be a different feature in Flow.